docs/csharp/tour-of-csharp/classes-and-objects.md

```diff
@@ -0,0 +1,728 @@
+
+### Events
+
+An ***event*** is a member that enables a class or object to provide notifications. An event is declared like a field except that the declaration includes an event keyword and the type must be a delegate type.
+
+Within a class that declares an event member, the event behaves just like a field of a delegate type (provided the event is not abstract and does not declare accessors). The field stores a reference to a delegate that represents the event handlers that have been added to the event. If no event handlers are present, the field is `null`.
+
+The `List<T>` class declares a single event member called `Changed`, which indicates that a new item has been added to the list. The Changed event is raised by the `OnChanged` virtual method, which first checks whether the event is `null` (meaning that no handlers are present). The notion of raising an event is precisely equivalent to invoking the delegate represented by the event—thus, there are no special language constructs for raising events.
+
+Clients react to events through ***event handlers***. Event handlers are attached using the `+=` operator and removed using the `-=` operator. The following example attaches an event handler to the `Changed` event of a `List<string>`.
+
+```csharp
+using System;
+class EventExample
+{
+	static int changeCount;
+	static void ListChanged(object sender, EventArgs e) 
+	{
+		changeCount++;
+	}
+	static void Main() 
+	{
+		List<string> names = new List<string>();
+		names.Changed += new EventHandler(ListChanged);
+		names.Add("Liz");
+		names.Add("Martha");
+		names.Add("Beth");
+		Console.WriteLine(changeCount);		// Outputs "3"
+	}
+}
+```
+
+For advanced scenarios where control of the underlying storage of an event is desired, an event declaration can explicitly provide `add` and `remove` accessors, which are somewhat similar to the `set` accessor of a property.
+
+### Operators
+
+An ***operator*** is a member that defines the meaning of applying a particular expression operator to instances of a class. Three kinds of operators can be defined: unary operators, binary operators, and conversion operators. All operators must be declared as `public` and `static`.
+
+The `List<T>` class declares two operators, `operator ==` and `operator !=`, and thus gives new meaning to expressions that apply those operators to `List` instances. Specifically, the operators define equality of two `List<T>` instances as comparing each of the contained objects using their Equals methods. The following example uses the `==` operator to compare two `List<int>` instances.
+
+```csharp
+using System;
+class OperatorExample
+{
+	static void Main() 
+	{
+		List<int> a = new List<int>();
+		a.Add(1);
+		a.Add(2);
+		List<int> b = new List<int>();
+		b.Add(1);
+		b.Add(2);
+		Console.WriteLine(a == b);  // Outputs "True" 
+		b.Add(3);
+		Console.WriteLine(a == b);  // Outputs "False"
+	}
+}
+```
```

Comment:
Is the List you're using here the one you showed before? I was wondering if we should start adding snippets for these examples so that they go through CI and so forth. (could be a later goal)
I'm also asking this because I couldn't use the classes as-is. For example, the List<T> code is missing using System or overriding GetHashCode and I get an exception when I run this code (it's in an infinite loop because of the == operator perhaps and the fact that the Equals method calls == again).